### ISSUE TITLE
'selector' field in JSON report may contain invalid/incomplete CSS selector

### ISSUE THREAD

**Issue by thomasfischer-his** [a11ywatch/kayle#11]
First, apologies if I report this issue at the wrong place!

I am running a11ywatch as a container as pulled from hub.docker.com and then issuing request for accessibility scans via carefully crafted HTTP POST requests to `http://localhost:3280/api/scan`. That is working so far and I get back a JSON document with the scan's results.
However, for *some but not all* issues reported under 'data' -> 'issues', the 'selector' contains invalid (or incomplete) CSS selectors, which makes locating and further processing of reported issues difficult.

For example, take the following HTML page:
```HTML
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en">
<head>
<meta charset="UTF-8" />
<title>Hello</title>
</head>
<body>
<p>Hello</p>
</body>
</html>
```

The resulting JSON document for the curl invocation of `curl --output output.json --location --request POST http://localhost:3280/api/scan --header 'Content-Type: application/json' --data-raw '{ "url": "http://www.example.com/", "runners": ["htmlcs","axe","custom"] }'` (assuming the HTML code above is served from 'www.example.com') is as follows:

```JSON
{
   "code" : 200,
   "data" : {
      "domain" : "www.example.com",
      "issues" : [
         {
            "code" : "landmark-one-main",
            "context" : "<html xmlns=\"http://www.w3.org/1999/xhtml\" lang=\"en\"><head>\n<meta charset=\"UTF-8\">\n<t...</html>",
            "message" : "Document should have one main landmark",
            "recurrence" : 0,
            "runner" : "axe",
            "selector" : ">:nth-child(1)>:nth-child(1)",
            "type" : "error",
            "typeCode" : 1
         },
         {
            "code" : "page-has-heading-one",
            "context" : "<html xmlns=\"http://www.w3.org/1999/xhtml\" lang=\"en\"><head>\n<meta charset=\"UTF-8\">\n<t...</html>",
            "message" : "Page should contain a level-one heading",
            "recurrence" : 0,
            "runner" : "axe",
            "selector" : ">:nth-child(1)>:nth-child(1)",
            "type" : "error",
            "typeCode" : 1
         },
         {
            "code" : "region",
            "context" : "<p>Hello</p>",
            "message" : "All page content should be contained by landmarks",
            "recurrence" : 0,
            "runner" : "axe",
            "selector" : "body>:nth-child(1)",
            "type" : "error",
            "typeCode" : 1
         }
      ],
      "issuesInfo" : {
         "accessScore" : 100,
         "errorCount" : 3,
         "issueMeta" : {
            "skipContentIncluded" : true
         },
         "issuesFixedByCdn" : 0,
         "noticeCount" : 0,
         "possibleIssuesFixedByCdn" : 0,
         "totalIssues" : 3,
         "warningCount" : 0
      },
      "lastScanDate" : "2023-10-03T11:33:17.753Z",
      "online" : true,
      "pageLoadTime" : {
         "duration" : 5,
         "durationFormated" : "Cached/Extremely Fast"
      },
      "url" : "https://www.example.com/"
   },
   "message" : "",
   "success" : true
}
```

The 'issue' with code 'region' has a valid 'selector' (`body>:nth-child(1)`), but neither the issues 'page-has-heading-one' nor 'landmark-one-main' have one as those two selectors start with `>`. This symbol denotes 'child combinator' and the 'parent' of this relation is missing.

Expected output: If a concrete element in the DOM tree can be identified as the culprit, the 'selector' field should unambiguously refer to it. If no such unique element exists, then either 'html', 'body', or 'head' should be stated as selector (whatever fits best for a given issue).

**Comment by j-mendez** [comment]
Hi @thomasfischer-his, thank you for the details. Going to move this over to the proper repo [kayle](https://github.com/a11ywatch/kayle).

**Comment by j-mendez** [comment]
@thomasfischer-his this is now fixed in `kayle@0.7.12`, if you are using the A11yWatch CLI to start the instance with docker locally use the `--upgrade` flag on start to get the latest image.

**Comment by thomasfischer-his** [comment]
I can confirm that the problem is gone, there is an empty string for those two cases.

However, wouldn't it be better to have a selector like 'html > body' for issues what are specific to the body (like missing landmarks), as opposed to issues in the header, for example?

**Comment by j-mendez** [comment]
@thomasfischer-his yeah I think it would. Feel free to add or re-open the issue etc.

**Comment by metsis** [comment]
We are still encountering this a lot, e.g.

```
{
"selector":
">:nth-child(1)",
"code":"WCAG2AA.Principle1.Guideline1_3.1_3_2.G57",
"type":"notice",
"typeCode":3,
"message":"Check that the content is ordered in a meaningful sequence when linearised, such as when style sheets are disabled.",
"runner":"htmlcs",
"recurrence":0
}
```

We are running `0.8.51`. Is this a bug or should it be assumed that these invalid selectors are tied to the "top element" aka html?
